rules/dead-code/src/Rector/MethodCall/RemoveEmptyMethodCallRector.php

```diff
@@ -79,12 +80,16 @@ public function refactor(Node $node): ?Node
     {
         /** @var Scope $scope */
         $scope = $node->var->getAttribute(AttributeKey::SCOPE);
-        /** @var ObjectType $type */
+        /** @var ObjectType|ThisType $type */
         $type = $scope->getType($node->var);
         if (! $type instanceof ObjectType) {
```

Comment:
Here the `TypeWithClassName` might be used instead, it covers all the types that have some class name

Reply:
I found that `$type` is got `ObjectType` as well on PropertyFetch, eg: `$this->foo->run()` instead of `$foo->run()`, which currently I skipped, I added failing fixture in latest commit.

Reply:
I've added handle `$node->var instanceof PropertyFetch`

Reply:
On call `static::$var->methodName()`, the scope type is `PHPStan\Type\MixedType`

Reply:
PropertyFetch may came from parent class, that maybe has parent class again, it seems need loop parser on class has extends

Reply:
The following method call from `$this` seems not detected as `PropertyFetch` if property defined in parent:

```diff
class ParentClass
{
    public $obj;

    public function __construct(EmptyMethod $obj)
    {
        $this->obj = $obj;
    }
}

class A extends ParentClass
{
    public function execute()
    {
-        $this->obj->run();
+        $this->obj->run();
    }
}
```

Reply:
Ok, so, `$node->var` instanceof `PropertyFetch` need to be checked first, updated fixture in latest commit.

Reply:
On `dead-code` set validate, it got error:

```bash
 Fatal error: Uncaught Rector\Core\Exception\ShouldNotHappenException:  
         Node "PhpParser\Node\Expr\MethodCall" is child of                      
         "PhpParser\Node\Stmt\If_", so it cannot be removed as it would break   
         PHP code. Change or remove the parent node instead. in                 
         /home/runner/work/rector/rector/packages/node-removal/src/BreakingRemov
         alGuard.php:25        
```

I try skip it in next commit.